airlock/business_logic.py

```diff
@@ -1808,8 +1835,8 @@ def update_file_in_request(
             type=AuditEventType.REQUEST_FILE_WITHDRAW,
             user=user,
             path=relpath,
-            group=request_file.group,
-            filetype=request_file.filetype.name,
+            group=old_group,
+            filetype=old_filetype.name,
         )
         filegroup_data = self._dal.delete_file_from_request(
```

Comment:
For withdrawing, haven't we already done this? Is it going to be confusing in the audit log that we withdraw a withdrawn file for the second time? And then the next audit log is to update it, but we're actually adding it, aren't we? Can you withdraw a file and then change your mind and add it again, without it having been changed?

Reply:
> For withdrawing, haven't we already done this?

Not in this case. 

If the request is PENDING, and you withdraw a file, it *is* deleted. But then if you try to add again the file we would run the add handler (not the add_withdrawn handler), because it doesn't exist on the request.

This is dealing with a request in RETURNED state - when you withdraw the file in that state it is marked as withdrawn, *not* deleted. Then if you add the file back again, we run the add_withdrawn handler which deletes it & then adds it.

> Is it going to be confusing in the audit log that we withdraw a withdrawn file for the second time?

Yes, although it does reflect what's happening in the code - in that there are actually two withdraw events in this scenario as described above (first marks as withdrawn, second deletes). The only way around this is to not emit an audit event at all for the second withdraw event in this scenario, which means holding the audit log a bit wrong, or pushing this code out of BLL & into DAL, which I think means holding that separation a bit wrong. 

> And then the next audit log is to update it, but we're actually adding it, aren't we?

Yes, that's true from the end user's point of view, I could change?

> Can you withdraw a file and then change your mind and add it again, without it having been changed?

Yep, that's what this code is trying to do.

Reply:
Probably OK for now - I think we had some bigger questions about what and how much we put in audit logs/activity logs